Tests/SmithyTestUtilTests/ResponseTestUtilTests/FloatingPointEquivalenceCheckerTests.swift

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import SmithyTestUtil
+import XCTest
+
+class FloatingPointEquivalenceCheckerTests : XCTestCase {
```

Comment:
These tests are great!  They look like they cover all our cases pretty exhaustively.

With the generic param types, doing both Float & Double is probably overkill, but might as well keep both.